packages/core/src/federation/registerModule.ts

```diff
@@ -2,6 +2,6 @@ import type { AbstractRuntime } from "../runtime/abstractRuntime.ts";
 
 export type ModuleRegisterFunction<TRuntime extends AbstractRuntime = AbstractRuntime, TContext = unknown> = (runtime: TRuntime, context?: TContext) => void;
 
-export function registerModule(register: ModuleRegisterFunction, runtime: AbstractRuntime, context?: unknown) {
-    register(runtime, context);
+export async function registerModule(register: ModuleRegisterFunction, runtime: AbstractRuntime, context?: unknown) {
```

Comment:
Should we consider renaming this method to `registerModuleAsync`?

Reply:
I don't know, while I understand why you are making this suggestion, in this case it's optional to provide an asynchronous function. Furthermore, while in the .NET ecosystem this is a widely used practice, in frontend this is not frequently applied. The asynchronous nature of a function is usually identified through it's return type (A `Promise`).

To provide somes examples, let's take React Router, a very popular third party used by Squide. Most of the API functions of React Router which can be asynchronous aren't suffixed with "Async":

- [lazy](https://reactrouter.com/en/main/route/lazy) returns a Promise but itsn't suffixed with "Async"
- [action](https://reactrouter.com/en/main/route/action) returns a Promise but itsn't suffixed with "Async"
- [loader](https://reactrouter.com/en/main/route/loader) returns a Promise but itsn't suffixed with "Async"